src/utilities/int_array.c

```diff
@@ -537,3 +537,191 @@ hypre_IntArrayNegate( hypre_IntArray *v )
 
    return hypre_error_flag;
 }
+
+/*--------------------------------------------------------------------------
+ * hypre_IntArraySegregateByValue
+ *
+ * This function segregates the indices of an array "v" based on
+ * specified values.
+ *
+ * Input parameters:
+ *   - num_values: number of unique values found in "v"
+ *   - values: array of size "num_values" containing a list of unique values
+ *   - sizes: array of size "num_values" containing the number of occurrences
+ *            in "v" of each value from "values".
+ *   - v: object of type hypre_IntArray containing values for categorization.
+ *
+ * Output parameter:
+ *   - w: object of type hypre_IntArrayArray containing pointers to "num_values"
+ *        arrays. Each array contains the set of indices in "v" that map to a
+ *        particular value of the array "values".
+ *
+ * Example:
+ *   Consider the following:
+ *   - v = {-1, -1, 1, -1, -1, 1, 1, -1}
+ *   - values = {-1, 1}
+ *   - num_values = 2
+ *
+ *   This function computes:
+ *   - w[0] = {0, 1, 3, 4, 7}
+ *   - w[1] = {2, 5, 6}
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+hypre_IntArraySegregateByValue( HYPRE_Int             num_values,
```

Comment:
Not sure about the name "Segregate".  Too many negative connotations.  Maybe just "Separate"?

Reply:
Yes, let me change it, thanks!

Reply:
Done in [c44f000](https://github.com/hypre-space/hypre/pull/1036/commits/c44f000f526305a2bc30247b697b07ded4e02aed)